DESIGN.md

```diff
@@ -0,0 +1,37 @@
+# Apollo Server design goals
+
+The goal of this project is to build a GraphQL server for Node.js that is simple, flexible, and performant. It is a Node.js GraphQL server built for production use.
```

Comment:
Actually for us that is a very important point.  We need a production viable Graphql server. To this point if problems are found in the Graphql js reference implementation will the Apollo server either have its own fork or its own implementation?